_adr/86/index.adoc

```diff
@@ -13,21 +13,88 @@ tags:
+
+Upon a successful authentication, MAS SSO provides a valid token back to the Kafka broker. Kafka brokers as a part of its authentication flow, upon a successful response, it introspects the token received and performs a custom claims check. It verifies that the "Canary Service account Client" belongs to the same organization as the creator or owner of the Kafka instance. Kafka brokers deem the authentication to be successful after this.
+
+image::adr-86-current-authentication-flow.png[Current authenticaion flow] 
+
 ## Proposed architecture
 
+When using sso.redhat.com as the single authentication post migration, KFM will create the Canary Service Account Clients in a "Special Management Organization". This would mean all the internal service accounts such as the Canary for all users (for all organizations) are created in this single organization that is managed and owned by Red hat. The reasons for this are, the limit on the number of service accounts an organization can create do not apply for this organization, and it provides more security by restricting these internal service accounts to the organization that is owned by red hat rather than the customer's organization.
+
+This would mean that all the Canary service accounts for all customers/users will be created/associated with this "Special Management Organization" when compared to the existing architecture where they were scoped to the user's organization. This requires some additional steps in authentical flow described in the current architecture section.
+
+- KFM when building the "CustomClaimCheck" includes the  Canary service account client ID "clientID" field as well
+
+```
+   - authentication:
+      customClaimCheck: '@.rh-org-id == ''testorgid''|| @.org_id == ''testorgid'' || @.clientId == ''canar-client-Id'''
+      tokenEndpointUri: >-
+            https://tokenendpointuri.com/token
+       ...
+    serviceAccounts:
+    - name: canary
+      password: **********
+      principal: canary-client-Id    
+```
+
+
+- Token Endpoint URI would now point to sso.redhat.com
+- The existing authentication flow up until receiving a valid token from the service mentioned in Token end point URI continues as it is.
+- During the Token introspection and performing the custom claim check validation, the following will be done:
+  - Verify that the "Canary service account Client" belongs to the "Special Management Org"
+  - Verify that the "ClientID" in the custom claim check matches the expected Canary client ID
```

Comment:
This doesn't match the proposed claim check above. I think it should continue verifying that the token belongs to the org that created the cluster -OR- verify that the ClientID is the canary client

Reply:
good catch Mike. I have updated the text and diagram to reflect the same. For canary the authn will be match the clientID since the orgID claims cannot be matched.

Reply:
@Rajagopalan-Ranganathan - regarding the custom claim check, should this doc still mention that the orgID check is present (for tokens other than the canary's)? I just want to confirm the intent. Right now in the diagram and in this list it looks like it will be removed.

Reply:
OrgID will be present for other checks, this is about the canary authn part alone. if required i can amend the text accordingly.

Reply:
Thanks for clarifying!